wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceManager.java

```diff
@@ -307,7 +324,9 @@ public WorkspaceImpl updateWorkspace(String id, Workspace update)
 
     WorkspaceImpl workspace = workspaceDao.get(id);
     if (workspace.getConfig() != null) {
-      workspace.setConfig(new WorkspaceConfigImpl(update.getConfig()));
+      WorkspaceConfigImpl configImpl = new WorkspaceConfigImpl(update.getConfig());
+      relativizeRecipeLinks(configImpl);
```

Comment:
can we please return config instance from this to be more explicit? Ideally new instance.